src/ctx.rs

```diff
@@ -82,17 +112,35 @@ pub(crate) fn get_app_config_test_instance() -> AppConfig {
             ProxyRoute {
                 inbound_route: String::from("/test"),
                 outbound_route: String::from("https://komodoplatform.com"),
+                proxy_type: ProxyType::JsonRpc,
```

Comment:
> here we should tell who we are going to proxy (e.g., proxy_type: "quicknode")

hmm may be then change `proxy_type` field name to `req_type`? as ...yeah `json_rpc` and `http_get` are about requests, I was feeling this param name may confuse.

As I understand "quicknode" is the one of the endpoints which we use, I suppose such thing is already configured in `inbound_route/outbound_route` fields. Currently `routes` in app config are actual params which tell who we are going to proxy.

My idea was to create `HttpGet` type which can be used for any `outbound_route` which supports it. The same currently is for `JsonRpc`, it can be used for any routes which accepts JSON-RPC method call.

> so we can manage and handle them independently without bloating out the core layer of proxy.

We dont have to do the changes in the code if we have new endpoint which needs Json-Rpc Method Call or HTTP GET requests. All you need is just update the app configuration file with new routes and provide necessary proxy/req type.

The changes in proxy code will needed if we want:
- to support new request type
- to support new blockchain

Please correct me if I didnt understand your suggestion.

Reply:
> The changes in proxy code will needed if we want:
> 
> * to support new request type
> * to support new blockchain
> 
> Please correct me if I didnt understand your suggestion.

So I am suggesting to avoid supporting request based proxy and also avoid changing the core layer depending on the target proxy. We should be able to proxy the request to its target regardless its method (GET, POST, DELETE, etc.). And keep all the endpoint related logics seperated in different crates (e.g., quicknode crate for quicknode, xy for the other, zt for another, etc.) so the core remains plain and simple and all application logic will be seperated and easily maintainable.

Reply:
Done https://github.com/KomodoPlatform/komodo-defi-proxy/pull/22/commits/1c600306d86c8fcb1f91760c165be5a6f3b081fd

Agree, separating feature logic makes it easier to develop validation middleware and proxy logic for different features. I moved quicknode and moralis into `proxy` crate in different modules.